cli/src/cores/models/crawled-asset.ts

```diff
@@ -0,0 +1,78 @@
+import * as fs from 'fs';
+import * as mime from 'mime-types';
+import { basename } from 'node:path';
+import * as path from 'path';
+import crypto from 'crypto';
+import { AssetTypeEnum } from 'src/api/open-api';
+
+export class CrawledAsset {
+  public path: string;
+
+  public assetType?: AssetTypeEnum;
+  public assetData?: Buffer;
+  public deviceAssetId?: string;
+  public fileCreatedAt?: string;
+  public fileModifiedAt?: string;
+  public fileExtension?: string;
+  public sidecarData?: Buffer;
+  public sidecarPath?: string;
+  public fileSize!: number;
+  public skipped = false;
+
+  constructor(path: string) {
+    this.path = path;
+  }
+
+  async readData() {
+    this.assetData = await fs.promises.readFile(this.path);
```

Comment:
This reads the whole file into memory, which is not going to work for large files. Maybe `assetData` can be a type stream, or whatever is returned from `fs.createReadStream`.

Reply:
Done